### ISSUE TITLE
OSAL should separate task allocation from task activation

### ISSUE THREAD

**Issue by jphickey** [nasa/osal#693]
**Is your feature request related to a problem? Please describe.**
Currently `OS_TaskCreate()` both allocates and activates a task.  However in embedded software it is often useful to pre-allocate a task - which creates the stack and ID and anything else that might fail - but hold it stopped in the kernel scheduler while other initialization is completed.  When the app is ready, it makes a separate call to activate the previously-allocated task.  

The benefit is that the activation step doesn't do any new allocation so it is very unlikely to fail (simplifying error clean up), as well as being fast to execute as it just unblocks the task.

Most RTOS's actually do it this way internally, where VxWorks uses `taskInit` followed by `taskActivate`, and RTEMS uses `rtems_task_create` followed by `rtems_task_start`.  Notably POSIX/pthreads does not separate these two actions, but easy to mimic/replicate via the new OS_WaitForStateChange() function.

**Describe the solution you'd like**
Add two new functions:
 - `int32 OS_TaskAllocate(...)`, similar to OS_TaskCreate but without starting the task
 - `int32 OS_TaskActivate(...)`, which starts the task

The existing `OS_TaskCreate()` becomes just a combination of the two.  Which it really as it is today anyway, but the two are coupled down in the `OS_TaskCreate_Impl` layer and therefore cannot be invoked separately in OSAL user applications.  This would bring it up one layer and (optionally) allow the app developer to leverage it, if they want.

**Additional context**
The "timebase" objects on VxWorks also would benefit from having a separate allocate/start. 

**Requester Info**
Joseph Hickey, Vantage Systems, Inc.

**Comment by jphickey** [comment]
In addition to separating the allocate + activate ops, this API should allow the user application to pass an opaque argument to the user-supplied function.  This would significantly benefit applications that use multiple "worker threads" where the code is the same, but operating on multiple objects in memory using one task for each object.

So the prototypes for these two proposed functions should be something like:

```
int32 OS_TaskAllocate(osal_id_t *task_id, const char *task_name, osal_stackptr_t stack_pointer,
                      size_t stack_size, osal_priority_t priority, uint32 flags);
int32 OS_TaskActivate(osal_id_t task_id, OS_ArgCallback_t function_pointer, void *arg);
```
